Added basic documentation to pendulum.py

abhisheknaik96 opened the pull request.

abhisheknaik96 committed:
Added basic documentation to pendulum.py

szepi1991 commented on docs/environments/pendulum.md:
````diff
@@ -1,4 +1,56 @@
+     theta'' = tau - mu * theta' - g * sin(theta),
+```
+where theta is the angle with the vertical, mu is the friction coefficient,
+tau is the torque, and g is the acceleration due to gravity.
````
Maybe add that the apostrophe means derivate (and '' the second derivative)?

abhisheknaik96 replied:
Will do, thanks.

szepi1991 commented on docs/environments/pendulum.md:
```diff
@@ -1,4 +1,56 @@
+3) There are only three discrete actions (apply a torque of -1, apply a torque
+of +1, and apply no-op) as opposed to a continuous torque value.
+4) The default reward function is implemented as a sparse reward, i.e. there
+is only a reward of 1 attained when the angle is in the region specified by
```
I think it's more direct to say "sparse reward, i.e. when the pendulum is near upright the reward is 1, otherwise 0."

abhisheknaik96 replied:
Sure, since I'm giving more details in the 'rewards' heading anyway.

szepi1991 commented on docs/environments/pendulum.md:
```diff
@@ -1,4 +1,56 @@
 # Pendulum
+| Overview          | Specification                                  |
+|-------------------|------------------------------------------------|
+| Observation Space | 3-d array: [cos(theta), sin(theta), theta_dot] |
```
The other environments document the ranges as well. cos and sin have implied ranges, but theta_dot doesn't.

Checking the code, this could be infinite? Or maybe the friction effectively limits it, but we don't know where?

abhisheknaik96 replied:
I actually wanted to know the bounds on the angular velocity myself for some purpose. The default bound (`_MAX_SPEED`) is `np.inf`. Empirically I've not seen it exceed 8. Perhaps one could analytically compute the maximum velocity using the differential equation of motion. 

For now I'll say that by default there is no bound on it?

szepi1991 reviewed (changes requested):
I left a few minor comments but this is great thank you! Very nicely written.

abhisheknaik96 committed:
Update pendulum.md

Updated the documentation to address the three review comments. Also used latex to encode the mathematical symbols.
